lib/galaxy/webapps/galaxy/config_schema.yml

```diff
@@ -930,22 +930,26 @@ mapping:
         required: false
         desc: |
           E-mail domains blacklist is used for filtering out users that are using
-          disposable email address during the registration.  If their address domain
-          matches any domain in the blacklist, they are refused the registration.
+          disposable email addresses at registration.  If their address domain matches any
+          domain on the list, they are refused registration.
 
-          Example value 'config/disposable_email_blacklist.conf'
+          Example value 'email_blacklist.conf'
+
+          The value of this option will be resolved with respect to <config_dir>.
 
       whitelist_file:
         type: str
-        default: disposable_email_whitelist.conf
-        path_resolves_to: config_dir
```

Comment:
`path_resolves_to` is not fully used yet, but why remove it?

Reply:
`path_resolves_to` requires a default value. If we specify a default value, the attribute will be set, and an error will be raised (or logged in the current version) when galaxy tries to open that file on startup (that was the bug in #9780). I think the problem with having a default value set for a path is that there is no way to know whether a file is missing because the user did not intend to use it, or because they did, chose to stick with the default, and then forgot to provide the file. In the first case we don't want an error, in the second case we do.

Reply:
Maybe `path_resolves_to` shouldn't require a default value? I think the only place where a `None` value would be a problem is https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/config/__init__.py#L241-L242 . Maybe these 2 lines could be removed?

Reply:
Maybe; I'm not sure. There might be corner cases where a default would be needed. I'll check/test this. (I'm mostly off today, so I might not get to it until tomorrow.)